thewalrus/quantum/means_and_variances.py

```diff
@@ -432,3 +433,33 @@ def photon_number_cumulant(mu, cov, modes, hbar=2):
         kappa += term
 
     return kappa
+
+
+def click_cumulant(mu, cov, modes, hbar=2):
+    r"""Calculates the click cumulant of the modes in the Gaussian state.
+
+    Args:
+        mu (array): length-:math:`2N` means vector in xp-ordering.
+        cov (array): :math:`2N\times 2N` covariance matrix in xp-ordering.
+        modes (list or array): list of modes.
+        hbar (float): value of hbar in the uncertainty relation.
+
+    Returns:
+        (float): the cumulant
+    """
+
+    modes = list(modes)  # turns modes from array to list if passed in as array
+    num_modes = len(mu) // 2
+    kappa = 0
+    for pi in partition(modes):
+        size = len(pi)
+        term = factorial(size - 1) * (-1) ** (size - 1)
+        for B in pi:
+            B = list(set(B))  # remove repetitions
```

Comment:
why would there be repetitions in B? This would only happen if there were repetitions in the `modes` arg, in which case i think the cumulant is zero? So this could be handled as an edge case? (e.g. `len(set(modes)) == len(modes)`)?

Reply:
This is solved, I think

Reply:
agreed!